### PR TITLE
Soda

### PR DESCRIPTION
**Pull request by sod-5364395**
Hi Sara,
Have a look at the code and feel free to write comments where you think that I should revise the code or should add something or whatever. 
Or if you think that no changes are needed you can close the pull request.

### CODE REVIEW

**Comment by eidelen** [comment]
Hi Sara and David,
I fixed your merge conflict.
@SaraSalvador1991: Before you continue your work on the main branch, you have work through David's pull request and merge his branch into the main branch. From that point on your code base is again alined.

File: classes/linked_list.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
+from typing import Any
+from classes.node import Node
+
+
```
Comment (eidelen) [inline classes/linked_list.py:4]: That's quite an impressing test class :)

**Review by eidelen — APPROVED** [review]

File: classes/linked_list.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+            return
+        node = Node(data)
+        if index == 0:
+            node.set_next(self.head)
+            self.head = node
+            self.size += 1
+        elif 0 < index < self.size:
+            iter_node = self.head
+            for _ in range(0, index - 1):
+                iter_node = iter_node.get_next()
```
Comment (SaraSalvador1991) [inline classes/linked_list.py:98]: I do not understand what is done here. you update the iter_node in the for loop and then you set the next node, right? but I do not understand node.set_next(iter_node.get_next())and iter_node.set_next(node)

File: classes/linked_list.py [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+            return
+        data_found = False
+        current_node = self.head
+        previous_node = self.head
+        index = 0
+        while current_node is not None and data_found is False:
+            if current_node.get_data() == data:
+                data_found = True
+                print(f"Removed data '{data}'.")
+                previous_node.set_next(current_node.get_next())
```
Comment (SaraSalvador1991) [inline classes/linked_list.py:121]: also here

File: classes/linked_list.py (hunk 2 above)
Comment (sod-5364395) [reply classes/linked_list.py:98]: image
I refer to the example image. If you want to insert the `node with value 15`, in the for loop you iterate through the linked list with the iter_node until you reach the `node with value 91`. The `node with value 23` is the next node of iter_node. With node.set_next(iter_node.get_next()) you set the `node with value 23` as the next node of the `node with value 15` (right red arrow in the image). With iter_node.set_next(node) you set the `node with value 15` as the next node of the `node with value 91` (left red arrow).

Comment (SaraSalvador1991) [reply classes/linked_list.py:98]: thank you David for the explanation and the scheme, it's clearer now :)

**Pull request merged**